concrete/authentication/concrete/controller.php

```diff
@@ -185,13 +185,14 @@ public function forgot_password()
                     throw new \Exception($token->getErrorMessage());
                 }
 
-                if (!$vs->email($em)) {
-                    throw new \Exception(t('Invalid email address.'));
+                $e = $this->app->make('error');
+                if (!$this->app->make(EmailValidator::class)->isValid($em, $e)) {
+                    throw new \Exception($e->toText());
```

Comment:
We should try to type these exceptions wherever possible (I know the old code had just `\Exception`. I think `\InvalidArgumentException` would be more relevant, and a subtype of that would be ideal. But this is a larger discussion for exceptions in c5

Reply:
I'd use (a subclass of) UserMessageException, since it's an error that can be safely displayed to end users

Reply:
:+1: We definitely need to standardize how we use exceptions and we need to build our our own exception types. But this is a topic for a different discussion